engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java

```diff
@@ -672,6 +675,67 @@ public void shouldSetVariablesFromMultipleInstructions() {
             tuple("bar", "true"));
   }
 
+  @Test
+  public void shouldTerminateAndActivateElementInTheSameScope() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .subProcess(
+                    "sp",
+                    sp ->
+                        sp.embeddedSubProcess().startEvent().userTask("A").userTask("B").endEvent())
+                .endEvent()
+                .done())
+        .deploy();
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var terminateElement =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId("A")
+            .getFirst();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .modification()
+        .terminateElement(terminateElement.getKey())
+        .activateElement("B")
+        .modify();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords()
+                .onlyEvents()
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("A")
+                .limit("A", ProcessInstanceIntent.ELEMENT_TERMINATED)
+                .toList())
```

Comment:
🔧 The assertion can take the Stream directly. There's no need to collect it to a List